### PR TITLE
[sw/test] Clean up the existing flash test.

### PR DESCRIPTION
**Pull request by mcy**

### CODE REVIEW

File: sw/device/tests/flash_ctrl/flash_test.c [hunk 1]
```diff
@@ -2,140 +2,155 @@
...
+    uint32_t *xs_ = (xs);            \
+    uint32_t *ys_ = (ys);            \
+    size_t len_ = (len);             \
+    for (int i = 0; i < len_; ++i) { \
+      CHECK(xs_[i] == ys_[i]);       \
+    }                                \
+  } while (false)
+
+#define CHECK_EQZ(x) CHECK((x) == 0);
+#define CHECK_NEZ(x) CHECK((x) != 0);
```
Comment (gkelly) [inline sw/device/tests/flash_ctrl/flash_test.c]: No semicolon in single-line macro definitions.

File: sw/device/tests/flash_ctrl/flash_test.c [hunk 2]
```diff
@@ -2,140 +2,155 @@
...
-  // attempt to erase bad page, should error
-  break_on_error(!flash_page_erase(bad_addr_start));
+int main(int argc, char **argv) {
+  uart_init(UART_BAUD_RATE);
+  flash_init_block();
 
-  // erase the good page
-  break_on_error(flash_page_erase(good_addr_start));
+  flash_cfg_bank_erase(FLASH_BANK_0, /*erase_en=*/true);
+  flash_cfg_bank_erase(FLASH_BANK_1, /*erase_en=*/true);
```
Comment (gkelly) [inline sw/device/tests/flash_ctrl/flash_test.c]: I strongly dislike inline comments in function calls.  Drop these here and then the `flash_ctrl` DIF can have proper types for them.

File: sw/device/tests/flash_ctrl/flash_test.c (hunk 1 above)
Comment (mcy) [reply sw/device/tests/flash_ctrl/flash_test.c]: Oops.

File: sw/device/tests/flash_ctrl/flash_test.c (hunk 2 above)
Comment (mcy) [reply sw/device/tests/flash_ctrl/flash_test.c]: I figure these will just get wiped out when the DIF gets written? For now I think we should keep them... it feels out of scope for this PR.

**Pull request merged**